pkg/controller/vitessshard/reconcile_tablets.go

```diff
@@ -360,3 +362,45 @@ func tabletAvailableStatus(resultBuilder *results.Builder, pod *corev1.Pod, read
 	resultBuilder.RequeueAfter(tabletAvailableTime - readyDuration)
 	return corev1.ConditionFalse
 }
+
+func (r *ReconcileVitessShard) checkPVCDiskUpdates(ctx context.Context, tabletSpec *vttablet.Spec, pod *corev1.Pod) {
+	// If no PVC is configured for this tablet pod, bail out.
+	if tabletSpec.DataVolumePVCSpec == nil {
+		return
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	namespacedClaim := apitypes.NamespacedName{
+		Namespace: pod.Namespace,
+		Name:      tabletSpec.DataVolumePVCName,
+	}
+
+	// If a matching PVC doesn't exist for this tablet pod, bail out.
+	err := r.client.Get(ctx, namespacedClaim, pvc)
+	if err != nil {
+		return
+	}
+
+	// If the matching PVC does not have the FileSystemResizePending condition, bail out.
+	if !checkPVCFileSystemResizeCondition(pvc) {
+		return
+	}
+
+	// If all checks pass, set the resize annotation.
+	requestedDiskQuantity := tabletSpec.DataVolumePVCSpec.Resources.Requests[corev1.ResourceStorage]
+	tabletSpec.Annotations[pvcDiskSizeAnnotation] = requestedDiskQuantity.String()
+}
+
+func checkPVCFileSystemResizeCondition (pvc *corev1.PersistentVolumeClaim) bool {
+	for _, condition := range pvc.Status.Conditions {
+		if condition.Type != corev1.PersistentVolumeClaimFileSystemResizePending {
+			continue
+		}
+
+		if condition.Status == corev1.ConditionTrue {
+			return true
+		}
```

Comment:
We don't need to keep scanning after we find the `condition.Type` we're looking for.

```suggestion
		return condition.Status == corev1.ConditionTrue
```

Reply:
Done.